src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -678,4 +630,77 @@ private function getSerializerContext(string $operationType, bool $denormalizati
 
         return $resourceMetadata->getItemOperationAttribute($operationName, $contextKey, null, true);
     }
+
+    private function addPaginationParameters(ResourceMetadata $resourceMetadata, string $operationName, \ArrayObject $pathOperation)
+    {
+        if (!$this->paginationEnabled || !$resourceMetadata->getCollectionOperationAttribute($operationName, 'pagination_enabled', true, true)) {
+            return;
+        }
+
+        $pathOperation['parameters'][] = $this->getPaginationParameters();
+
+        if ($resourceMetadata->getCollectionOperationAttribute($operationName, 'pagination_client_items_per_page', $this->clientItemsPerPage, true)) {
+            $pathOperation['parameters'][] = $this->getItemsPerPageParameters();
+        }
+    }
+
+    /**
+     * @throws ResourceClassNotFoundException
+     */
+    private function addSubresourceOperation(array $subresourceOperation, \ArrayObject $definitions, string $operationId, array $mimeTypes, string $resourceClass, ResourceMetadata $resourceMetadata): \ArrayObject
+    {
+        $operationName = 'get';
+        $subResourceMetadata = $this->resourceMetadataFactory->create($subresourceOperation['resource_class']);
+        $serializerContext = $this->getSerializerContext(OperationType::SUBRESOURCE, false, $subResourceMetadata, $operationName);
+        $responseDefinitionKey = $this->getDefinition($definitions, $subResourceMetadata, $subresourceOperation['resource_class'], $serializerContext);
+
+        $pathOperation = new \ArrayObject([]);
+        $pathOperation['tags'] = $subresourceOperation['shortNames'];
+        $pathOperation['operationId'] = $operationId;
+        $pathOperation['produces'] = $mimeTypes;
+        $pathOperation['summary'] = sprintf('Retrieves %s%s resource%s.', $subresourceOperation['collection'] ? 'the collection of ' : 'a ', $subresourceOperation['shortNames'][0], $subresourceOperation['collection'] ? 's' : '');
+        $pathOperation['responses'] = $this->getSubresourceResponse($subresourceOperation['collection'], $subresourceOperation['shortNames'][0], $responseDefinitionKey);
+
+        // Avoid duplicates parameters when there is a filter on a subresource identifier
+        $parametersMemory = [];
+        $pathOperation['parameters'] = [];
+
+        foreach ($subresourceOperation['identifiers'] as list($identifier, , $hasIdentifier)) {
+            if (true === $hasIdentifier) {
+                $pathOperation['parameters'][] = ['name' => $identifier, 'in' => 'path', 'required' => true, 'type' => 'string'];
+                $parametersMemory[] = $identifier;
+            }
+        }
+
+        if ($parameters = $this->getFiltersParameters($resourceClass, $operationName, $subResourceMetadata, $definitions, $serializerContext)) {
```

Comment:
With **$resourceClass**, it fails on entities with composite key, with **$subresourceOperation['resource_class']** it goes correctly.

```suggestion
        if ($parameters = $this->getFiltersParameters($subresourceOperation['resource_class'], $operationName, $subResourceMetadata, $definitions, $serializerContext)) {
```